core/state.go

```diff
@@ -437,3 +451,135 @@ func (s *State) ContractIsAlreadyDeployedAt(addr *felt.Felt, blockNumber uint64)
 	}
 	return deployedAt <= blockNumber, nil
 }
+
+func (s *State) Revert(blockNumber uint64, update *StateUpdate) error {
+	err := s.expectRoot(update.NewRoot)
+	if err != nil {
+		return err
+	}
```

Comment:
```suggestion
	if err := s.expectRoot(update.NewRoot); err != nil {
		return err
	}
```

Reply:
This is actually intentional, it let's us reuse err variable throughout the entire functional call.